lib/src/simulations/epicycloid_curve.dart

```diff
@@ -0,0 +1,385 @@
+  double total;
+  final bool animatefactor;
+  final bool animatepoints;
+  final bool animating;
+
+  @override
+  _EpicycloidState createState() => _EpicycloidState();
+}
+
+class _EpicycloidState extends State<Epicycloid> {
+  nextStep() {
+    setState(() {
+      sleep(Duration(milliseconds: 10));
+
+      if (widget.animatefactor) {
+        widget.factor += 0.01;
+      }
+      if (widget.animatepoints) {
+        widget.total += 0.3;
+      }
+
+      if (widget.factor > 51) {
+        widget.factor = 0;
+      }
+      if (widget.total > 500) {
+        widget.total = 0;
+      }
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (widget.animating) {
+      WidgetsBinding.instance.addPostFrameCallback((_) {
+        nextStep();
+      });
+    }
+    return Stack(
+      alignment: Alignment.center,
+      children: <Widget>[
+        CustomPaint(
+          painter: EpicycloidPainter(
+              widget.factor,
+              widget.total,
+              (MediaQuery.of(context).size.width / 2.4).roundToDouble(),
+              (MediaQuery.of(context).size.width / 2).roundToDouble(),
+              (MediaQuery.of(context).size.height / 3).roundToDouble(),
+              widget.animatefactor,
+              widget.animatepoints,
+              Theme.of(context).accentColor),
+          child: Container(),
+        ),
+        Visibility(
+          visible: widget.animatepoints,
+          child: Positioned(
+            bottom: 60,
+            child: Text("Points: ${widget.total.toInt()}"),
+          ),
+        ),
+        Visibility(
+          visible: widget.animatefactor,
+          child: Positioned(
+            bottom: 40,
+            child: Text("Factor: ${widget.factor.toStringAsFixed(1)}"),
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+class EpicycloidPainter extends CustomPainter {
+  List<Offset> points = [];
+  double total, factor;
+  double radius, tx, ty;
+  bool animatefactor, animatepoints;
+  Color color;
+
+  EpicycloidPainter(
+    this.factor,
+    this.total,
+    this.radius,
+    this.tx,
+    this.ty,
+    this.animatefactor,
+    this.animatepoints,
+    this.color,
+  );
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    var paint = Paint();
+    paint.color = Colors.red;
+    paint.strokeWidth = 3;
+    paint.style = PaintingStyle.stroke;
+
+    var paint2 = Paint();
+    paint2.color = color;
+    paint2.strokeWidth = 1;
+    double x = 2 * pi / total;
+    double angle = pi;
+
+    for (int i = 0; i < total; ++i) {
+      points.add(
+          Offset(radius * cos(angle), radius * sin(angle)).translate(tx, ty));
+      angle = angle + x;
+    }
+
+    for (double i = 0; i < total; i += 1) {
+      canvas.drawLine(points[(i % total).toInt()],
+          points[((i * factor) % total).toInt()], paint2);
+    }
+
+    canvas.drawCircle(Offset(tx, ty), radius, paint);
+  }
+
+  @override
+  bool shouldRepaint(EpicycloidPainter oldDelegate) => true;
```

Comment:
I think this would work with false, you'll have to try.

Reply:
It works with a false value but then when the slider is updated, the curve doesn't get updated with it. Also the animation is not as smooth as before. Leaving it true is gives a better result!